src/math/power.c

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Intel Corporation. All rights reserved.
+//
+// Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+//
+//
+
+ /* Include Files */
+#include <sof/math/power.h>
+
+/* Function Definitions */
+/**
+ * Arguments	: int64_t denominator
+ *
+ * Return Type	: int32_t
+ */
+static inline int32_t div_s32s64(int64_t denominator)
+{
+	uint64_t b_denominator;
+	int32_t quotient;
+
+	if (denominator == 0) {
+		quotient = INT32_MAX;
+	} else {
+		b_denominator =
+		    (denominator < 0) ? ~(uint64_t)denominator + 1 : (uint64_t)denominator;
+		b_denominator = B_NUMERATOR / b_denominator;
+
+		if (denominator < 0)
+			/* for negative denominator, typecast uint64_t to int32_t for */
+			/* negative quotient */
+			quotient = (int32_t)-(int64_t)b_denominator;
+		else
+			quotient = (int32_t)b_denominator;
+	}
+	return quotient;
+}
+
+/**
+ * Arguments	: int32_t b
+ * base input range [-32 to +32]
+ *
+ *		: int32_t e
+ * exponent input range [ -3 to +3 ]
+ *
+ * Return Type	: int32_t
+ * power output range [-32768 to + 32768]
+ *
+ * +------------------+------------------+------------------+--------+--------+--------+
+ * | base	      | exponent	 | power(returntype)|	b    |	e     |	 p     |
+ * +----+-----+-------+----+-----+-------+----+----+--------+--------+--------+--------+
+ * |WLen| FLen|Signbit|WLen| FLen|Signbit|WLen|FLen|Signbit | Qformat| Qformat| Qformat|
+ * +----+-----+-------+----+-----+-------+----+----+--------+--------+--------+--------+
+ * | 32 | 25  |	 1    | 32 | 29	 |  1	 | 32 | 15 |   1    | 6.26   | 2.30   | 16.16  |
+ * +------------------+------------------+------------------+--------+--------+--------+
+ */
+int32_t power_scalar_function(int32_t b, int32_t e)
```

Comment:
Just thinking why so long name, this is in float math library just pow() or powf(). E.g. pow_int32()?

Reply:
OK , I will replace with `power_int32()`